contracts/proposal/dao-proposal-single/src/contract.rs

```diff
@@ -255,6 +258,77 @@ pub fn execute_propose(
         .add_attribute("status", proposal.status.to_string()))
 }
 
+pub fn execute_veto(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    proposal_id: u64,
+) -> Result<Response, ContractError> {
+    let mut prop = PROPOSALS
+        .may_load(deps.storage, proposal_id)?
+        .ok_or(ContractError::NoSuchProposal { id: proposal_id })?;
+
+    match prop.status {
+        Status::Timelocked { expires } => {
+            match prop.timelock {
+                Some(ref timelock) => {
```

Comment:
Alternatively, the nesting can be reduced with `if let`, e.g.:

`if let Status::Timelocked { expires } = prop.status {
    return Err(ContractError::NotPassed {});
}`

Which would also have a benefit of handling the error sooner.

Reply:
Sorry, got it wrong above, but could be refactored like this to reduce one level of nesting.

` let expires = match prop.status {
        Status::Timelocked { expires } => expires,
        _ => return Err(ContractError::NotPassed {})
    };`